lib/matplotlib/tests/test_colors.py

```diff
@@ -1734,3 +1734,29 @@ def test_colorizer_vmin_vmax():
     assert ca.vmax == 3.0
     assert ca.norm.vmin == 1.0
     assert ca.norm.vmax == 3.0
+
+
+def test_lsc_from_list_color_alpha_tuple():
+    """
+    GitHub issue #29042: A bug in 'from_list' causes an error
+    when passing a tuple (str, float) where the string is a
+    color name or grayscale value and float is an alpha value.
+    """
+    valid_colors = [("red", 0.3), ("0.42", 0.1), "green"]
+    mcolors.LinearSegmentedColormap.from_list("testcolormap", valid_colors)
+
+
+@pytest.mark.parametrize("colors",
+                         [[(0.42, "blue"), (.1, .1, .1, .1)],
+                          ["blue", (0.42, "red")],
+                          ["blue", (.1, .1, .1, .1), ("red", 2)],
+                          [(0, "red"), (1.1, "blue")],
+                          [(0.52, "red"), (0.42, "blue")]])
+def test_lsc_from_list_invalid_inputs(colors):
+    with pytest.raises(ValueError):
+        mcolors.LinearSegmentedColormap.from_list("testcolormap", colors)
+
+
+def test_lsc_from_list_value_color_tuple():
+    value_color_tuples = [(0.42, "red"), (0.6, "blue"), (1, "green")]
+    mcolors.LinearSegmentedColormap.from_list("testcolormap", value_color_tuples)
```

Comment:
Similar to the above case, please assert the color values trough calling `cmap(...)`. You might want to tune `N` and the position values so that they land exactly on lookup table entries. Alternatively, make an approximate check with `nptest.assert_allclose()`.